src/rest/json.cpp

```diff
@@ -39,6 +40,16 @@ namespace otbr {
 namespace rest {
 namespace Json {
 
+std::string Bytes2HexString(const uint8_t *aBytes, uint8_t aLength)
+{
+    char hex[2 * aLength + 1];
```

Comment:
not c++ standard?
```suggestion
    char* hex = new char[2 * aLength + 1];
```

Reply:
I am not a C++ expert, but looking at [Array declaration on cppreference.com](https://en.cppreference.com/w/cpp/language/array) it doesn't look like anything would forbid this type of declaration?

In any case, the function just below uses the same type of declaration. If we should get rid of this type of declaration we probably should do it there too.

In general json.cpp probably could use some modernization using new C++ constructs.